nservicebus/cancellation-and-exception-handling.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Cancellation and exception handling
+summary: Using Azure for endpoint hosting and to provide Transports and Persistence
+reviewed: 2021-05-26
+---
+
+In most cases, code in NServiceBus handlers and sagas should not catch exceptions, and prefer instead allowing [recoverability](/nservicebus/recoverability/) to deal with exceptions using retries.
+
+However, when it is necessary to handle exceptions, care should be taken so that [cooperative cancellation](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/task-cancellation) will work correctly when the endpoint needs to be shut down.
+
+All NServiceBus components use the [Particular.Analyzers](https://www.myget.org/feed/particular/package/nuget/Particular.Analyzers) package which contains Roslyn analyzers to verify the patterns described here, among others. If developing an extension to NServiceBus, using the same analyzers package is highly recommended.
+
+## Catching `Exception`
+
+One common pattern is to catch and swallow the general `Exception` type in some cases, like this:
+
+```csharp
+try
+{
+    await SomeOperation(cancellationToken);
```

Comment:
I know what we mean but I wonder if users will treat this as "all tokens no matter what" rather than "if this is the token that NServiceBus provides to participate in shutdown", do you think readers will get that or could we make it more clear somehow?

Reply:
Putting inside a handler looked too "you should catch exceptions inside a handler" and I just couldn't bring myself to do it, so I tried to get the job done with expanded comments in the snippet.